Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -174,6 +175,30 @@ std::map<std::string, std::string> DiscusMultipleScatteringCorrection::validateI
   return issues;
 }
 
+void DiscusMultipleScatteringCorrection::getXMinMax(const Mantid::API::MatrixWorkspace &ws, double &xmin,
+                                                    double &xmax) const {
+  // set to crazy values to start
+  xmin = std::numeric_limits<double>::max();
+  xmax = -1.0 * xmin;
+  size_t numberOfSpectra = ws.getNumberHistograms();
+
+  // determine the data range - only return min > 0. Bins with x=0 will be skipped later on
+  for (size_t workspaceIndex = 0; workspaceIndex < numberOfSpectra; workspaceIndex++) {
+    const auto &dataX = ws.points(workspaceIndex);
```

Comment:
Do we need to add a check here fore is monitor/has detector?

Reply:
I think those would be useful checks so I'll add them in